sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/non_streaming_order_by_aggregator.py

```diff
@@ -0,0 +1,152 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 Microsoft Corporation
+
+"""Internal class for multi execution context aggregator implementation in the Azure Cosmos database service.
+"""
+
+from azure.cosmos._execution_context.aio.base_execution_context import _QueryExecutionContextBase
+from azure.cosmos._execution_context.aio.multi_execution_aggregator import _MultiExecutionContextAggregator
+from azure.cosmos._execution_context.aio import document_producer
+from azure.cosmos._routing import routing_range
+from azure.cosmos import exceptions
+
+# pylint: disable=protected-access
+
+class _NonStreamingOrderByContextAggregator(_QueryExecutionContextBase):
+    """This class is a subclass of the query execution context base and serves for
+    non-streaming order by queries. It is very similar to the existing MultiExecutionContextAggregator,
+    but is needed since we're dealing with items and not document producers.
+
+    This class builds upon the multi-execution aggregator, building a document producer per partition
+    and draining their results entirely in order to create the result set relevant to the filters passed
+    by the user.
+    """
+
+    def __init__(self, client, resource_link, query, options, partitioned_query_ex_info):
+        super(_NonStreamingOrderByContextAggregator, self).__init__(client, options)
+
+        # use the routing provider in the client
+        self._routing_provider = client._routing_map_provider
+        self._client = client
+        self._resource_link = resource_link
+        self._query = query
+        self._partitioned_query_ex_info = partitioned_query_ex_info
+        self._sort_orders = partitioned_query_ex_info.get_order_by()
+        self._orderByPQ = _MultiExecutionContextAggregator.PriorityQueue()
```

Comment:
Not sure whether this is usual in python - but I find it super weird to reuse a subclass of another class - in Java or .Net the usual approach would be to put the child class (PriorityQueue) into its own file/class?

Reply:
It's unusual in Python as well.
It is a pattern, but if a class is localized to within another class, I wouldn't expect it to then also be used externally. So the structure here certainly feels strange.